sphinx_gallery/directives.py

```diff
@@ -67,12 +70,16 @@ def run(self):
         # Retrieve the backreferences directory
         config = self.state.document.settings.env.config
         backreferences_dir = config.sphinx_gallery_conf["backreferences_dir"]
+        if backreferences_dir is None:
+            logger.warning(
```

Comment:
Also possibly silly question, but I note a `self.warning` (using `Directive.warning`) used above - in what case should we be using the `Directive` warning and in what case should we be using our logger?

I tried `self.warning` and it did not output to the build. Where does the warning output to..?